src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -1102,6 +1320,43 @@ public sealed class StopComputerCommand : PSCmdlet, IDisposable
         private readonly CancellationTokenSource _cancel = new CancellationTokenSource();
         private const int forcedShutdown = 5; // See https://msdn.microsoft.com/library/aa394058(v=vs.85).aspx
 
+        private const uint PlannedShutdown = 0x80000000;
+        private const uint Other = 0x00000000;
+
+        // Major Shutdown Reasons
+        private const uint Application = 0x00040000;
+        private const uint Hardware = 0x80010000;
+        private const uint OperatingSystem = 0x80020000;
+        private const uint Power = 0x80060000;
+        private const uint Software = 0x80030000;
+        private const uint System = 0x80050000;
+
+        // Minor Shutdown Reasons
+        private const uint BlueScreen = 0x0000000F;
+        private const uint Disk = 0x00000007;
+        private const uint Environment = 0x0000000c;
+        private const uint Driver = 0x0000000d;
+        private const uint HotFix = 0x00000011;
+        private const uint HotFixUninstall = 0x00000017;
+        private const uint Unresponsive = 0x00000005;
+        private const uint Installation = 0x00000002;
+        private const uint Maintenance = 0x00000001;
+        private const uint MMC = 0x00000019;
+        private const uint NetworkConnectivity = 0x00000014;
+        private const uint NetworkCard = 0x00000009;
+        private const uint OtherDriver = 0x0000000e;
+        private const uint PowerSupply = 0x0000000a;
+        private const uint Processor = 0x00000008;
+        private const uint Reconfigure = 0x00000004;
+        private const uint SecurityIssue = 0x00000013;
+        private const uint SecurityPatch = 0x00000012;
+        private const uint SecurityPatchUninstallation = 0x00000018;
+        private const uint ServicePack = 0x00000010;
+        private const uint ServicePackUninstallation = 0x00000016;
+        private const uint TerminalServices = 0x00000020;
+        private const uint Unstable = 0x00000006;
+        private const uint Upgrade = 0x00000003;
+        private const uint WMI = 0x00000015;
```

Comment:
Same as above; it looks like you could roll most or all of these into a single enum that is shared between both commands.